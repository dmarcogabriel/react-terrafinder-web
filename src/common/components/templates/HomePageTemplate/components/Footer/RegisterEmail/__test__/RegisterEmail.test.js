import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitFor } from '@testing-library/react';
import { RegisterEmail } from '..';

jest.mock('hooks/useNotification', () => ({
  useNotification: () => ({
    showNotification() {},
  }),
  NOTIFICATION_TYPES: { SUCCESS: 'success', ERROR: 'error' },
}));

describe('<RegisterEmail />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<RegisterEmail />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should send email correctly', () => {
    const { getByTestId } = render(<RegisterEmail />);

    const registerEmail = getByTestId('registerEmail');
    const emailInput = getByTestId('emailInput');

    fireEvent.change(emailInput, {
      target: { value: 'email@test.com.br' },
    });
    fireEvent.click(registerEmail);

    expect(emailInput).toHaveValue('email@test.com.br');
  });

  it('should show error on invalid', async () => {
    const { getByTestId, rerender } = render(<RegisterEmail />);

    const emailInput = getByTestId('emailInput');
    const submitBtn = getByTestId('registerEmail');

    fireEvent.change(emailInput, {
      target: { value: 't' },
    });
    fireEvent.click(submitBtn);

    await waitFor(() => rerender(<RegisterEmail />));

    expect(emailInput).toHaveValue('t');
    expect(emailInput.parentElement).toHaveClass('Mui-error');
    expect(submitBtn).toHaveClass('Mui-disabled');
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<RegisterEmail />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
