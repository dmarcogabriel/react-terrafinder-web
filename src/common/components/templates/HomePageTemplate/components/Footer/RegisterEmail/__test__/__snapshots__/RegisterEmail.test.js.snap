// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterEmail /> matches snapshot 1`] = `
<form
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1waotp6-MuiPaper-root"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-13bnupw-MuiInputBase-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
      data-testid="emailInput"
      inputMode="email"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value=""
    />
  </div>
  <button
    className="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-ke5b6m-MuiButtonBase-root-MuiButton-root"
    data-testid="registerEmail"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Cadastrar
  </button>
</form>
`;
